classes/curlmanager_security_helper.php

```diff
@@ -59,64 +59,64 @@ private function log_curl_http_requests(string $urlstring) : bool {
             return true;
         }
 
-        // Just return false if we are not enabled - no blocking.
-        if (!get_config('tool_curlmanager', 'enabled')) {
-            return false;
-        }
-
         // Check if the host is in allowed list.
         $returnvalue = $this->host_is_allowed($host);
 
         // Call moodle curl_security_helper method url_is_blocked.
         $curlsecurityhelper = new curl_security_helper();
         $urlblocked = $curlsecurityhelper->url_is_blocked($urlstring);
 
-        $rootcodepath = '';
-        $trace = debug_backtrace();
-        $formattedbacktrace = format_backtrace(debug_backtrace(), true);
-        $lasttrace = count($trace) - 1;
-        if (isset($trace[$lasttrace]['file'])) {
-            $rootcodepath = $trace[$lasttrace]['file'];
-        }
-
-        $plugin = $this->getcomponentbycodepath($rootcodepath);
-        if ($plugin === false) {
-            $plugin = '';
-        }
+        // At Some Point TM this will need to be changed when this gets true unit tests.
+        // For now, let's stop it from busting core tests.
+        $log = get_config('tool_curlmanager', 'loggingenabled') && !PHPUNIT_TEST;
```

Comment:
how about an upgrade step to clone the config into both new settings?

Reply:
The default behaviour just gives you this for free at upgrade time. The core problem here is that  it should be logging and not blocking unless disabled. The problem here is more with my original fix

Reply:
but using the default behaviour already in place is a 1:1 no behaviour change